### PR TITLE
Scheduler config

### PR DESCRIPTION
**Pull request by olpipi**

### CODE REVIEW

File: src/cpp/include/openvino/genai/llm_pipeline.hpp [hunk 1]
```diff
@@ -105,7 +106,8 @@ class OPENVINO_GENAI_EXPORTS LLMPipeline {
     LLMPipeline(
         const std::string& path, 
         const std::string& device="CPU", 
-        const ov::AnyMap& plugin_config={}
+        const ov::AnyMap& plugin_config={},
+        OptionalSchedulerConfig scheduler_config
```
Comment (ilya-lavrenov) [inline src/cpp/include/openvino/genai/llm_pipeline.hpp:110]: can scheduler be passed as plugin_config option?
Then in CB pipeline, we can extract this option and remaining ones will be passed to compile_model

File: src/cpp/src/llm_pipeline.cpp
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:1]: could you please also drop current way with CB as device name?

Comment (olpipi) [reply src/cpp/src/llm_pipeline.cpp:1]: it is already done. device = "CB" is not supported anymore

File: src/cpp/src/llm_pipeline.cpp [hunk 2]
```diff
@@ -385,23 +385,25 @@ class ContinuousBatchingAdapter final : public LLMPipelineImplBase {
         const std::filesystem::path& model_path,
         const Tokenizer& tokenizer,
         const std::string& device,
-        const ov::AnyMap& plugin_config
+        const ov::AnyMap& plugin_config,
+        const SchedulerConfig& scheduler_config
```
Comment (Wovchena) [inline src/cpp/src/llm_pipeline.cpp:389]: `scheduler_config` is ignored with the new approach. Remove it

File: src/cpp/include/openvino/genai/generation_config.hpp [hunk 3]
```diff
@@ -142,6 +142,8 @@ static constexpr ov::Property<float> presence_penalty{"presence_penalty"};
 static constexpr ov::Property<float> frequency_penalty{"frequency_penalty"};
 static constexpr ov::Property<size_t> rng_seed{"rng_seed"};
 
+static constexpr ov::Property<SchedulerConfig> enable_continuous_batching{"enable_continuous_batching"};
```
Comment (ilya-lavrenov) [inline src/cpp/include/openvino/genai/generation_config.hpp:145]: why is  part of generation_config.hpp ? I suppose llm_pipeline is the right place.

File: src/cpp/include/openvino/genai/scheduler_config.hpp [hunk 4]
```diff
@@ -5,6 +5,8 @@
 
 #include <cstddef>
 
+#include "openvino/runtime/properties.hpp"
```
Comment (ilya-lavrenov) [inline src/cpp/include/openvino/genai/scheduler_config.hpp:8]: is it required in this file?

File: tests/python_tests/ov_genai_test_utils.py [hunk 5]
```diff
@@ -223,4 +222,5 @@ def load_pipe(configs: List[Tuple], temp_path):
 
 @functools.lru_cache(1)
 def get_continuous_batching(path):
-    return ov_genai.LLMPipeline(str(path), ov_genai.Tokenizer(str(path)), 'CB')
+    default_scheduler_config = ov_genai.SchedulerConfig
+    return ov_genai.LLMPipeline(str(path), ov_genai.Tokenizer(str(path)), 'CPU', config={"enable_continuous_batching": default_scheduler_config})
```
Comment (ilya-lavrenov) [inline tests/python_tests/ov_genai_test_utils.py:226]: do we have python bindings for properties of generation_config ? if yes, let's implement binding for enable_continuous_batching as well

File: src/cpp/src/llm_pipeline.cpp [hunk 6]
```diff
@@ -523,9 +525,10 @@ ov::genai::LLMPipeline::LLMPipeline(
     const std::string& device,
     const ov::AnyMap& plugin_config
 ){
-    auto start_time = std::chrono::steady_clock::now();    
-    if ("CB" == device) {
-        m_pimpl = std::make_unique<ContinuousBatchingAdapter>(model_path, tokenizer, "CPU", plugin_config);
+    auto start_time = std::chrono::steady_clock::now();
+    if (auto scheduler_config_prop = plugin_config.find(ov::genai::enable_continuous_batching.name()) != plugin_config.end()) {
+        auto scheduler_config = plugin_config.at(ov::genai::enable_continuous_batching.name());
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline.cpp:530]: why variable is called scheduler_config, while it's actually enable_continuous_batching ?

File: src/cpp/src/llm_pipeline.cpp (hunk 2 above)
Comment (Wovchena) [reply src/cpp/src/llm_pipeline.cpp:389]: A possibility to provide `SchedulerConfig` as `plugin_config` should be described in docstrings. I forgot to do that when I added CB device.

File: tests/python_tests/ov_genai_test_utils.py [hunk 7]
```diff
@@ -223,4 +222,5 @@ def load_pipe(configs: List[Tuple], temp_path):
 
 @functools.lru_cache(1)
 def get_continuous_batching(path):
-    return ov_genai.LLMPipeline(str(path), ov_genai.Tokenizer(str(path)), 'CB')
+    default_scheduler_config = ov_genai.SchedulerConfig
```
Comment (ilya-lavrenov) [inline tests/python_tests/ov_genai_test_utils.py:225]: SchedulerConfig is a class, not an object
Looks like you need to create instance of this class here.

File: src/cpp/include/openvino/genai/llm_pipeline.hpp [hunk 8]
```diff
@@ -22,6 +22,8 @@ using OptionalGenerationConfig = std::optional<GenerationConfig>;
 using EncodedInputs = std::variant<ov::Tensor, TokenizedInputs>;
 using StringInputs = std::variant<std::string, std::vector<std::string>>;
 
+static constexpr ov::Property<SchedulerConfig> scheduler_config{"scheduler_config"};
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/cpp/include/openvino/genai/llm_pipeline.hpp:26]: done

File: src/cpp/include/openvino/genai/generation_config.hpp [hunk 9]
```diff
@@ -141,7 +141,6 @@ static constexpr ov::Property<int64_t> eos_token_id{"eos_token_id"};
 static constexpr ov::Property<float> presence_penalty{"presence_penalty"};
 static constexpr ov::Property<float> frequency_penalty{"frequency_penalty"};
 static constexpr ov::Property<size_t> rng_seed{"rng_seed"};
-
```
[the remark this answers is not in the document]
Comment (olpipi) [reply src/cpp/include/openvino/genai/generation_config.hpp]: done

File: tests/python_tests/ov_genai_test_utils.py (hunk 7 above)
Comment (olpipi) [reply tests/python_tests/ov_genai_test_utils.py]: done

File: tests/python_tests/ov_genai_test_utils.py [hunk 10]
```diff
@@ -223,4 +222,5 @@ def load_pipe(configs: List[Tuple], temp_path):
 
 @functools.lru_cache(1)
 def get_continuous_batching(path):
-    return ov_genai.LLMPipeline(str(path), ov_genai.Tokenizer(str(path)), 'CB')
+    default_scheduler_config = ov_genai.SchedulerConfig()
```
[the remark this answers is not in the document]
Comment (olpipi) [reply tests/python_tests/ov_genai_test_utils.py]: There is a check in ContinuousBatchingPipeling. 
Set cache_size=1 as before

File: src/cpp/include/openvino/genai/llm_pipeline.hpp [hunk 11]
```diff
@@ -22,6 +23,8 @@ using OptionalGenerationConfig = std::optional<GenerationConfig>;
...
```
Comment (Wovchena) [inline src/cpp/include/openvino/genai/llm_pipeline.hpp:26]: Please, add docstrings to .hpp and python bindings constructors about Continuous batching activation. Openvino docs is the primary source of knowledge. We can influence openvino docs via these docstrings

Comment (Wovchena) [inline src/cpp/include/openvino/genai/llm_pipeline.hpp:26]: Is it actually needed? `Property` was added to enable positional arguments. But `ov::AnyMap` is used to pass `SchedulerConfig` now. I suspect `AnyMap` should be able to store `SchedulerConfig` without `ov::Property` definition.

File: tests/python_tests/ov_genai_test_utils.py [hunk 12]
```diff
@@ -13,7 +13,6 @@
 import shutil
 import json
 
-
```
Comment (Wovchena) [inline tests/python_tests/ov_genai_test_utils.py:16]: Put it back :) https://peps.python.org/pep-0008/ recommends to have two empty lines.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: src/cpp/include/openvino/genai/llm_pipeline.hpp (hunk 11 above)
Comment (Wovchena) [reply src/cpp/include/openvino/genai/llm_pipeline.hpp:26]: Never mind

**Review by ilya-lavrenov — APPROVED** [review]

File: src/python/py_generate_pipeline.cpp [hunk 13]
```diff
@@ -503,17 +505,28 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
...
         py::arg("model_path"),
         py::arg("tokenizer"),
         py::arg("device") = "CPU",
+        py::arg("config") = ov::AnyMap({}), "openvino.properties map",
         R"(
             LLMPipeline class constructor for manualy created openvino_genai.Tokenizer.
             model_path (str): Path to the model file.
             tokenizer (openvino_genai.Tokenizer): tokenizer object.
             device (str): Device to run the model on (e.g., CPU, GPU). Default is 'CPU'.
+            Add {"scheduler_config": ov_genai.SchedulerConfig} to config properties to create continuous batching pipeline.
```
Comment (Wovchena) [inline src/python/py_generate_pipeline.cpp:529]: Please, add it to C++ constructors as well to keep them aligned

**Review by Wovchena — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**